studio/components/layouts/ProjectLayout/LayoutHeader/ProjectDropdown.tsx

```diff
@@ -15,9 +15,16 @@ import { IS_PLATFORM, PROJECT_STATUS } from 'lib/constants'
 // highest common route with just projectRef in the router queries.
 
 const sanitizeRoute = (route: string, routerQueries: ParsedUrlQuery) => {
-  let queryArray = Object.entries(routerQueries)
+  const queryArray = Object.entries(routerQueries)
+
   if (queryArray.length > 1) {
-    return route.split('/').slice(0, 4).join('/')
+    // account for query string, if exists (example: /logs/explorer?q=select...)
+    const hasQueryString = queryArray.filter(([key]) => key === 'q').length > 0
+
+    return route
+      .split('/')
+      .slice(0, hasQueryString ? 5 : 4)
```

Comment:
nit: hard coded numbers feel brittle... if we are removing the last directory in the path, might consider:
```suggestion
      .slice(0, -1)
```

Reply:
Yeah it is brittle Greg, but not sure of a way around it. The other alternative is baking logic in for just the logs explorer pages with query strings. 

Your .slice(0, -1) works but still doesn't account for when logs explorer pages have query strings:

![CleanShot 2023-05-23 at 15 39 54@2x](https://github.com/supabase/supabase/assets/105593/5e28a2a4-edf1-450a-b704-c504fcf4c5cb)